packages/providers/src/lib/push/apns/apns.provider.ts

```diff
@@ -73,4 +73,20 @@ export class APNSPushProvider extends BaseProvider implements IPushProvider {
       date: new Date().toISOString(),
     };
   }
+
+  private validateAndFormatKey(key: string): string {
+    // Check if key is already properly formatted
+    const properFormat = /^-----BEGIN PRIVATE KEY-----\n[A-Za-z0-9+/=\n]+\n-----END PRIVATE KEY-----$/;
+
+    if (properFormat.test(key)) {
+      return key; // Key is already in correct format
+    }
+
+    // If not properly formatted, clean and format the key
+    const cleanKey = key.replace(/-----BEGIN PRIVATE KEY-----|-----END PRIVATE KEY-----|[\s\n\r]/g, '');
+
+    const formattedKey = cleanKey.match(/.{1,64}/g)?.join('\n') || '';
+
+    return `-----BEGIN PRIVATE KEY-----\n${formattedKey}\n-----END PRIVATE KEY-----`;
```

Comment:
As discussed offline, we will add a validation error during the creation of APN integration so that users are informed proactively about invalid integration properties.